README.md
# **Ubuntu-Setup**

![Shell](https://img.shields.io/static/v1?label=Shell%20Script&logo=Ubuntu&message=Linux&color=orange)
![GitHub](https://img.shields.io/github/license/fstecnologiainfo/Ubuntu-Setup)
![GitHub top language](https://img.shields.io/github/languages/top/fstecnologiainfo/Ubuntu-Setup?logo=Linux&logoColor=white)
![GitHub repo size](https://img.shields.io/github/repo-size/fstecnologiainfo/Ubuntu-Setup)

 Script para realizar instalações e customização do Ubuntu Server e Desktop com versões superior a 21.10, incluído distros instaladas no Microsoft WSL2.

>**IMPORTANTE**\
>Esses scripts são programados para executar apenas 1 vez, caso haja algum erro verifique os arquivos de log em **/var/log/** \
>Caso queira realizar modificação nos programas que serão instalados, modifique a sessão de **VARIAVEIS GLOBAIS**

***

## **COMO USAR?**

1 - Faça o download do arquivo compactado na página do GitHub ou clone o repositório usando o comando

Git
```
git clone https://github.com/fstecnologiainfo/Ubuntu-Setup.git
```

PowerShell
```
Invoke-WebRequest -Uri https://codeload.github.com/fstecnologiainfo/Ubuntu-Setup/zip/refs/heads/main -OutFile C:\Ubuntu-Setup.zip
```

2 - Acesse a pasta **/Ubuntu-Setup** e altere as permissões do arquivos *setup-(tipo-distro).sh* em arquivos executáveis.

3 - Execute a instalação com os privilégios *root*, sendo **/Ubuntu-Setup/setup-serve.sh** para realizar as instalações e customização no Ubuntu Server, **/Ubuntu-Setup/setup-workstation.sh** para realizar as instalações e customização no Ubuntu Desktop e **/Ubuntu-Setup/setup-wsl2.sh** para realizar as instalações e customização no Microsoft WSL2 Ubuntu.

***

## **SERVER**

***Instalação***

Em desenvolvimento

***

## **WORKSTATION**

***Instalação***

Softwares SNAP: *Whatsdesk - MIcrosoft Teams - PDF Mix Tool*

Softwares .DEB: *htop - VLC - FileZilla - Flatpak - Git - Preload - Curl - qBittorrent - Nano - Nomacs*

Softwares .DEB (Administrador de Redes): *Synaptic Open VPN - Gnome Boxes - Remmina - Net-tools - DNS Utils - Neofetch - SpeedTest (cli) - NMap - Open SSL*

***Customização***

- Ajuste de data e hora para dual boot e VM's
- Codecs para execução de arquivos de media

***

## **MICROSOFT WSL2**

***Instalação***

Softwares .DEB (Administrador de Redes): *NMap - SpeedTest (cli) - Neofetch - Net-tools - cMatrix - Figlet - Open SSL - DNS  Utils*

***Extra***

ACME: Para gerar certificados com Let's Encrypt

OpenSSL: Para gerar certificados auto-assinados e certificados .pfx

***

## **CRÉDITOS**

**Autor**: Fernando da S. Conceição

**Contato**: [GMail](fstecnologia.info@gmail.com)

CHANGELOG
# CHANGE  LOG

Criador: **Fernando da Silva Conceição**\
Data de criação do arquivo: **10/04/2022**\
Ultima Modificação: **30/04/2022**
***

## 30/04/2022

Modificações

arquivo: **README.md**

arquivo: **setup-workstation.sh**

- otimizando códigos e saídas dos logs
- verificações se as instalações de SNAP e DEB funcionam corretamente
- mudando condição de verificação do usuário root
- ajustando formatação do código

arquivo: **setup-workstation.sh**

- mudando condição de verificação iniciais
- realizando ajuste de LOG
- ajustando formatação do código

arquivo: **setup-server.sh**

Exclusão

arquivo: **parametros.sh**

***

## 26/04/2022

Modificações

arquivo: **README.md**

arquivo: **/ssh/banner.net**

***

## 24/04/2022

Modificações

arquivo: **setup-workstation.sh**

- Melhoria no arquivo de Log detalhado
- Removendo instalação do GRUB-THEME

arquivo: **parametros.sh**

- Adicionando funções para impressão de Banner de identificação de execução do script
- Alterando programas SNAP a serem instalados

***

## 18/04/2022

Criação

- Adicionando

arquivo **setup-wsl2.sh** para instalação para Microsoft WSL2

Modificações

arquivo: **parametros.sh**

- Adicionando variáveis para instalações do WSL2
- Alterando variáveis de instalação de Workstation

arquivo: **README.md**

***

## 17/04/2022

Modificações

arquivo: **setup-workstation.sh**

- Corrigindo mensagens no terminal
- Ajustando lógica de verificação da versão do sistema operacional
- Ajustando LOGS

arquivo: **parametros.sh**

- Alterando saídas de log

***

## 16/04/2022

Criação

- Adicionando pasta /conf/Grub-Theme e arquivos para implementação do mesmo

Modificações

arquivo: **parametros.sh**

- Adicionando variáveis Globais para logs, usuário e versionamento do sistema operacional

arquivo: **setup-workstation.sh**

- Adicionando funções de instalação dos software SNAP e DEB
- Adicionando funções para cálculos de tempo de execução
- Adicionando funções para customização do sistema

arquivo: **.gitignore**

arquivo: **README.md**

***

## 10/04/2022

Criação

- Criação do arquivo CHANGELOG para informar alterações no projeto
- Criação do arquivo parametro.sh, setup-server.sh e setup-workstation.sh
- Criação da pasta conf para armazenar arquivos de configurações e personalizações dos programas

Modificações

- Personalização do arquivo *\conf\neofetch\config.conf
- Personalização do Arquivo *\conf\ssh\banner.net

***

## 15/03/2022

Criação

- Criação do projeto e arquivos padrões do GIT


parametros.sh
#!/bin/bash
# Desenvolvido: Fernando da Silva Conceição
# Data de criação do arquivo: 10/04/2022
# Ultima Modificação: 24/04/2022
# Versão: 0.4
#============================================================================================

# VARIAVEIS GLOBAIS
# Utilizados para todos os scripts
LOGSCRIPT="/var/log/$(echo $0 | cut -d'/' -f2).log"
LOGSCRIPT2="/var/log/SIMPLE-$(echo $0 | cut -d'/' -f2).log"
USUARIO=$(id -u)
UBUNTU=$(lsb_release -rs)

# VARIAVEIS SERVER

# VARIAVEIS WORKSTATION

# Utilizados como flags para exucutar ou não a instalações dos recursos
# As respostas devem ser SIM ou NAO
SNAP_RESP='SIM'
DEB_RESP='SIM'
DEBADM_RESP='SIM'

WSNAP='whatsdesk teams-for-linux pdfmixtool'
WDEB='htop vlc filezilla flatpak git preload curl qbittorrent nano grub-customizer nomacs ubuntu-restricted-extras'
WDEBADM='synaptic openvpn gnome-boxes remmina net-tools dnsutils neofetch speedtest-cli nmap openssl'

# VARIAVEIS WSL2
WSL2DEB='nmap speedtest-cli neofetch net-tools cmatrix figlet openssl dnsutils'

#============================================================================================
# BANNERS
bannerDesk(){
echo -e " ____  _____ ____  _  _______ ___  ____ "
echo -e "|  _ \| ____/ ___|| |/ /_   _/ _ \|  _ \ "
echo -e "| | | |  _| \___ \| ' /  | || | | | |_) | "
echo -e "| |_| | |___ ___) | . \  | || |_| |  __/ "
echo -e "|____/|_____|____/|_|\_\ |_| \___/|_| "
echo -e "============================================"
echo -e ""
}

bannerServer(){
echo -e " ____  _____ ______     _______ ____ "
echo -e "/ ___|| ____|  _ \ \   / / ____|  _ \ "
echo -e "\___ \|  _| | |_) \ \ / /|  _| | |_) | "
echo -e " ___) | |___|  _ < \ V / | |___|  _ < "
echo -e "|____/|_____|_| \_\ \_/  |_____|_| \_\ "
echo -e "============================================"
echo -e ""
}

bannerWSL(){
echo -e " __  __ _                           __ _    __        ______  _     ____ "
echo -e "|  \/  (_) ___ _ __ ___  ___  ___  / _| |_  \ \      / / ___|| |   |___ \ "
echo -e "| |\/| | |/ __| '__/ _ \/ __|/ _ \| |_| __|  \ \ /\ / /\___ \| |     __) | "
echo -e "| |  | | | (__| | | (_) \__ \ (_) |  _| |_    \ V  V /  ___) | |___ / __/ "
echo -e "|_|  |_|_|\___|_|  \___/|___/\___/|_|  \__|    \_/\_/  |____/|_____|_____| "
echo -e "============================================================================"
echo -e ""
}

setup-server.sh
#!/bin/bash
# Desenvolvido: Fernando da Silva Conceição
# Data de criação do arquivo: 10/04/2022
# Ultima Modificação: 30/04/2022
# Versão: 0.1
#============================================================================================

# IMPORTAÇÃO DOS PARAMETROS e VARIAVEIS GLOBAIS
source parametros.sh
LOG_DETAIL=$LOGSCRIPT
LOG=$LOGSCRIPT2
LOCAL=$(pwd)
VERSAO="1.3"
#============================================================================================
#* APRESENTACAO DO SCRIPT
echo -e " ____  _____ ______     _______ ____ "
echo -e "/ ___|| ____|  _ \ \   / / ____|  _ \ "
echo -e "\___ \|  _| | |_) \ \ / /|  _| | |_) | "
echo -e " ___) | |___|  _ < \ V / | |___|  _ < "
echo -e "|____/|_____|_| \_\ \_/  |_____|_| \_\ "
echo -e "============================================"
echo -e "Versão do Script: $VERSAO"
echo -e ""
#============================================================================================
# VERIFICAÇÕES INICIAIS

setup-workstation.sh
#!/bin/bash
# Desenvolvido: Fernando da S. Conceição
# Data de criação do arquivo: 10/04/2022
# Ultima Modificação: 30/04/2022
# Versão: 1.3
#TODO Testar a não interação do usuário
#TODO Implementar os FLAGS para instalação
#TODO Testar echo com saidas para log para que sejam exibidas no terminal
#============================================================================================
#* VARIAVEIS GLOBAIS
LOGD="/var/log/$(echo $0 | cut -d'/' -f2)-DETALHADO.log"
LOG="/var/log/$(echo $0 | cut -d'/' -f2)-RESUMIDO.log"
LOCAL=$(pwd)
VERSAO="1.3"
WSNAP='whatsdesk teams-for-linux pdfmixtool'
WDEB='htop vlc filezilla flatpak git preload curl qbittorrent nano nomacs ubuntu-restricted-extras'
WDEBADM='synaptic openvpn gnome-boxes remmina net-tools dnsutils neofetch speedtest-cli nmap openssl'
#============================================================================================
#* APRESENTACAO DO SCRIPT
clear
echo -e " ____  _____ ____  _  _______ ___  ____ "
echo -e "|  _ \| ____/ ___|| |/ /_   _/ _ \|  _ \ "
echo -e "| | | |  _| \___ \| ' /  | || | | | |_) | "
echo -e "| |_| | |___ ___) | . \  | || |_| |  __/ "
echo -e "|____/|_____|____/|_|\_\ |_| \___/|_| "
echo -e "============================================"
echo -e "Versão do Script: $VERSAO"
echo -e ""

#============================================================================================
#* VERIFICAÇÕES INICIARIS
echo -e "Verificando se o usuario e Root..."
if [ "$(id -u)" -ne "0" ]
	then
		echo -e "Usuário $(whoami) não é Root"
		echo -e "Caso você não tenha executado o script com o comando: sudo"
		echo -e "Execute novamente o script para verificar o ambiente."
        exit 1
	else
		echo -e "O usuário é Root, continuando com o script..."
		sleep 5
fi

echo -e "Verificando se o script já foi executado..."
if [ -f $LOG ]
	then
		# populando LOG com informações do script e execução
		echo "Data de Execução: $(date)" >> $LOGD
		echo "Local de execução do script: $LOCAL" >> $LOGD
		echo "Usuario: $(whoami) " >> $LOGD
		echo "Script executado: $(echo $0 | cut -d'/' -f2)" >> $LOGD
		echo "Versão do Script: $VERSAO" >> $LOGD
		echo "======================================================" >> $LOGD

		echo "Data de Execução: $(date)" >> $LOG
		echo "Local de execução do script: $LOCAL" >> $LOG
		echo "Usuario: $(whoami) " >> $LOG
		echo "Script executado: $(echo $0 | cut -d'/' -f2)" >> $LOG
		echo "Versão do Script: $VERSAO" >> $LOG
		echo "======================================================" >> $LOG
		echo -e "Script $0 já foi executado 1 (uma) vez nesse computador..."
		echo -e "É recomendado analisar o arquivo de $LOG e $LOGD para informações de falhas ou erros"
		echo -e "Todos os scripts foram projetados para serem executados apenas 1 (uma) vez."
		echo "Nâo passou pela checagem de execução unica." >> $LOG
		sleep 5
		exit 1
	else
		# populando LOG com informações do script e execução
		echo "Data de Execução: $(date)" >> $LOGD
		echo "Local de Execução: $LOCAL" >> $LOGD
		echo "Usuario: $(whoami) " >> $LOGD
		echo "Script executado: $(echo $0 | cut -d'/' -f2)" >> $LOGD
		echo "Versão do Script: $VERSAO" >> $LOGD
		echo "======================================================" >> $LOGD

		echo "Data de Execução: $(date)" >> $LOG
		echo "Local de Execução: $LOCAL" >> $LOG
		echo "Usuario: $(whoami) " >> $LOG
		echo "Script executado: $(echo $0 | cut -d'/' -f2)" >> $LOG
		echo "Versão do Script: $VERSAO" >> $LOG
		echo "======================================================" >> $LOG
		echo -e "Primeira vez que você está executando esse script, tudo OK!"
		echo -e "Agora tome um café e espere a magica acontecer..."
		echo "Passou pela checagem de execução unica." >> $LOG
		sleep 5
fi

echo -e "Verificando compatibilidade com o sistema operacional..."
if [ "$(lsb_release -rs)" == "21.10" ] || [ "$(lsb_release -rs)" == "22.04" ]
	then
		echo -e "A versão do sistema operacional é superior a 21.10, continuando com o script..."
		echo "Passou pela checagem de versão do sistema operacional." >> $LOG
		sleep 5
	else
		echo -e "O sistema operacional não é compativel com o script."
		echo "Não passou pela checagem do sistema operacional." >> $LOG
		exit 1
fi

echo -e "Verificando conexão com internet..."
if [ "$(nc -zw1 google.com 443 &> /dev/null ; echo $?)" == "0" ]
	then
		echo -e "Você tem acesso a Internet, continuando com o script..."
		sleep 5
	else
		echo -e "Você NÃO tem acesso a Internet, verifique suas configurações de rede IPV4 e conexões e execute novamente este script."
		echo "Não passou pela checagem de conxão com a internet." >> $LOG
		sleep 5
		exit 1
fi

echo "Passou pelas verificações iniciais!" &>> $LOGD
sleep 5

#============================================================================================
#* AJUSTES DO SISTEMA
echo -e "Iniciando customização e ajustes do sistema..."
echo "Alterando sistema de data e hora..." >> $LOGD
timedatectl set-local-rtc 1 --adjust-system-clock &>> $LOGD

#============================================================================================
#* ATUALIZAÇÃO DO SISTEMA
HORAINICIAL=$(date +%T)
HORA_UP=$(date +%T)
echo -e "Atualizando repositórios..."
echo "Atualizando repositórios..." >> $LOG
echo "Atualizando repositórios..." >> $LOGD
if ! sudo apt-get update &>> $LOGD
then
    echo -e "Não foi possível atualizar os repositórios. Verifique seu arquivo /etc/apt/sources.list"
    echo "Não foi possível atualizar os repositórios. Verifique seu arquivo /etc/apt/sources.list" >> $LOG
    exit 1
fi
echo -e "Atualização do repositório feita com sucesso!!"

echo -e "Atualizando pacotes e sistema..."
echo "Atualizando pacotes e sistema..." >> $LOG
echo "Atualizando pacotes e sistema..." >> $LOGD
if ! sudo apt-get upgrade --autoremove -y &>> $LOGD
then
    echo -e "Não foi possível atualizar pacotes."
    echo "Não foi possível atualizar pacotes." >> $LOG
    exit 1
fi
echo -e "Atualização de pacotes feita com sucesso!!"
HORA_UPFIM=$(date +%T)

#============================================================================================
#* INSTALAÇÂO DE SNAPs
HORA_SNAP=$(date +%T)
echo -e "Instalando suporte a SNAP e aplicativos SNAP..."
echo "Instalando suporte a SNAP e aplicativos SNAP..." >> $LOG
echo "Instalando suporte a SNAP..." >> $LOGD
if ! sudo apt install snap -y &>> $LOGD
then
    echo -e "Não foi possível instalar o suporte a snap."
    echo "Não foi possível instalar o suporte a snap." >> $LOG
	echo -e "Verifique o LOG em $LOGD para maires detalhes.."
    exit 1
fi
echo -e "Instalação do suporte a SNAP feita com sucesso!!"

echo -e "Instalando softwares SNAPs..."
echo "Instalando softwares SNAPs..." >> $LOG
echo "Instalando softwares SNAPs..." >> $LOGD
if ! sudo snap install $WSNAP &>> $LOGD
then
	echo -e "Não foi possivel realizar a instalação dos aplicativos SNAP."
	echo "Não foi possivel realizar a instalação dos aplicativos SNAP." >> $LOG
	echo -e "Verifique o LOG em $LOGD para maires detalhes.."
	exit 1
fi
echo "Instalação dos aplicativos SNAP feita com sucesso!!"
HORA_SNAPFIM=$(date +%T)

#============================================================================================
#* INSTALAÇÂO DE SOFTWARES .DEB
HORA_DEB=$(date +%T)
echo -e "Instalando aplicativos DEB..."
echo "Instalando aplicativos DEB..." >> $LOG
echo "Instalando aplicativos DEB..." >> $LOGD
if ! sudo DEBIAN_FRONTEND=noninteractive apt install $WDEB -y &>> $LOGD
then
	echo -e "Não foi possivel instalar todos os pacotes DEB."
	echo "Não foi possivel instalar todos os pacotes DEB." >> $LOG
	echo -e "Verifique o LOG em $LOGD para maiores detalhes.."
	exit 1
fi
echo -e "Instalação dos pacotes DEB finalizados com sucesso!!"

#============================================================================================
#* INSTALAÇÂO DE SOFTWARES .DEB ADMINISTRATIVOS
echo -e "Instalando aplicativos DEB administrativos..."
echo "Instalando aplicativos DEB administrativos..." >> $LOG
echo "Instalando aplicativos DEB administrativos..." >> $LOGD
if ! sudo DEBIAN_FRONTEND=noninteractive apt install $WDEBADM -y &>> $LOGD
then
	echo -e "Não foi possivel instalar todos os pacotes .DEB admnistrativos."
	echo "Não foi possivel instalar todos os pacotes .DEB admnistrativos." >> $LOG
	echo -e "Verifique o LOG em $LOGD para maires detalhes.."
	exit 1
fi
echo -e "Instalação dos pacotes administrativos concuida com sucesso!!"
HORA_DEBFIM=$(date +%T)

HORAFINAL=$(date +%T)

#============================================================================================
#* CALCULOS DOS TEMPOS GASTOS DE EXECUÇÃO
HORAUP01=$(date -u -d "$HORA_UP" +"%s")
HORAUP02=$(date -u -d "$HORA_UPFIM" +"%s")
TEMPO_UP=$(date -u -d "0 $HORAUP01 sec - $HORAUP02 sec" +"%H:%M:%S")
echo -e "Tempo gasto para atualização do sistema: $TEMPO_UP" >> $LOG
echo "Tempo gasto para atualização do sistema: $TEMPO_UP" &>> $LOGD

HORASNAP01=$(date -u -d "$HORA_SNAP" +"%s")
HORASNAP02=$(date -u -d "$HORA_SNAPFIM" +"%s")
TEMPO_SNAP=$(date -u -d "0 $HORASNAP01 sec - $HORASNAP02 sec" +"%H:%M:%S")
echo -e "Tempo gasto instalar softwares SNAP: $TEMPO_SNAP" >> $LOG
echo "Tempo gasto instalar softwares SNAP: $TEMPO_SNAP" &>>LOGD

HORADEB01=$(date -u -d "$HORA_DEB" +"%s")
HORADEB02=$(date -u -d "$HORA_DEBFINAL" +"%s")
TEMPO_DEB=$(date -u -d "0 $HORADEB01 sec - $HORADEB02 sec" +"%H:%M:%S")
echo -e "Tempo gasto instalar softwares .DEB: $TEMPO_DEB" >> $LOG
echo "Tempo gasto instalar softwares .DEB: $TEMPO_DEB" &>> $LOGD

HORAINICIAL01=$(date -u -d "$HORAINICIAL" +"%s")
HORAFINAL01=$(date -u -d "$HORAFINAL" +"%s")
TEMPO=$(date -u -d "0 $HORAFINAL01 sec - $HORAINICIAL01 sec" +"%H:%M:%S")
echo -e "Tempo gasto para execução do script $0: $TEMPO" >> $LOG
echo "Tempo gasto para execução do script $0: $TEMPO" &>> $LOGD
#============================================================================================
#* FIM DO PROCESSO E RESTART
echo -e "Pressione <Enter> para concluir o processo e reiniciar o sistema."
read
echo -e "Fim do script $0 em: $(date +%d/%m/%Y-"("%H:%M")")\n" >> $LOGD
echo -e "Reiniciando sistema para aplicar atualizações!....." >> $LOG
sleep 10
sudo reboot now
exit 1

setup-wsl2.sh
#!/bin/bash
# Desenvolvido: Fernando da S. Conceição
# Data de criação do arquivo: 18/04/2022
# Ultima Modificação: 30/04/2022
# Versão: 0.2
#TODO Implementar os FLAGS para instalação
#TODO Mudar echos com saidas para log para que sejam exibidas no terminal
#============================================================================================
#* VARIAVEIS GLOBAIS
HORAINICIAL=$(date +%T)
LOGD="/var/log/$(echo $0 | cut -d'/' -f2)-DETALHADO.log"
LOG="/var/log/$(echo $0 | cut -d'/' -f2)-RESUMIDO.log"
LOCAL=$(pwd)
VERSAO="0.2"
WSL2DEB='nmap speedtest-cli neofetch net-tools cmatrix figlet openssl dnsutils'
#============================================================================================
#* APRESENTACAO DO SCRIPT
echo -e " __  __ _                           __ _    __        ______  _     ____ "
echo -e "|  \/  (_) ___ _ __ ___  ___  ___  / _| |_  \ \      / / ___|| |   |___ \ "
echo -e "| |\/| | |/ __| '__/ _ \/ __|/ _ \| |_| __|  \ \ /\ / /\___ \| |     __) | "
echo -e "| |  | | | (__| | | (_) \__ \ (_) |  _| |_    \ V  V /  ___) | |___ / __/ "
echo -e "|_|  |_|_|\___|_|  \___/|___/\___/|_|  \__|    \_/\_/  |____/|_____|_____| "
echo -e "============================================================================"
echo -e "Versão do Script: $VERSAO"
echo -e ""

#============================================================================================
#* VERIFICAÇÕES INICIAIS
echo -e "Verificando se o usuario e Root..."
if [ "$(id -u)" -ne "0" ]
	then
		echo -e "Usuário $(whoami) não é Root"
		echo -e "Caso você não tenha executado o script com o comando: sudo"
		echo -e "Execute novamente o script para verificar o ambiente."
        exit 1
	else
		echo -e "O usuário é Root, continuando com o script..."
		sleep 5
fi

echo -e "Verificando se o script já foi executado..."
if [ -f $LOG ]
	then
		# populando LOG com informações do script e execução
		echo "Data de Execução: $(date)" >> $LOGD
		echo "Local de execução do script: $LOCAL" >> $LOGD
		echo "Usuario: $(whoami) " >> $LOGD
		echo "Script executado: $(echo $0 | cut -d'/' -f2)" >> $LOGD
		echo "Versão do Script: $VERSAO" >> $LOGD
		echo "======================================================" >> $LOGD

		echo "Data de Execução: $(date)" >> $LOG
		echo "Local de execução do script: $LOCAL" >> $LOG
		echo "Usuario: $(whoami) " >> $LOG
		echo "Script executado: $(echo $0 | cut -d'/' -f2)" >> $LOG
		echo "Versão do Script: $VERSAO" >> $LOG
		echo "======================================================" >> $LOG
		echo -e "Script $0 já foi executado 1 (uma) vez nesse computador..."
		echo -e "É recomendado analisar o arquivo de $LOG e $LOGD para informações de falhas ou erros"
		echo -e "Todos os scripts foram projetados para serem executados apenas 1 (uma) vez."
		echo "Nâo passou pela checagem de execução unica." >> $LOG
		sleep 5
		exit 1
	else
		# populando LOG com informações do script e execução
		echo "Data de Execução: $(date)" >> $LOGD
		echo "Local de Execução: $LOCAL" >> $LOGD
		echo "Usuario: $(whoami) " >> $LOGD
		echo "Script executado: $(echo $0 | cut -d'/' -f2)" >> $LOGD
		echo "Versão do Script: $VERSAO" >> $LOGD
		echo "======================================================" >> $LOGD

		echo "Data de Execução: $(date)" >> $LOG
		echo "Local de Execução: $LOCAL" >> $LOG
		echo "Usuario: $(whoami) " >> $LOG
		echo "Script executado: $(echo $0 | cut -d'/' -f2)" >> $LOG
		echo "Versão do Script: $VERSAO" >> $LOG
		echo "======================================================" >> $LOG
		echo -e "Primeira vez que você está executando esse script, tudo OK!"
		echo -e "Agora tome um café e espere a magica acontecer..."
		echo "Passou pela checagem de execução unica." >> $LOG
		sleep 5
fi

echo -e "Verificando compatibilidade com o sistema operacional..."
if [ "$(lsb_release -rs)" == "21.10" ] || [ "$(lsb_release -rs)" == "22.04" ]
	then
		echo -e "A versão do sistema operacional é superior a 21.10, continuando com o script..."
		echo "Passou pela checagem de versão do sistema operacional." >> $LOG
		sleep 5
	else
		echo -e "O sistema operacional não é compativel com o script."
		echo "Não passou pela checagem do sistema operacional." >> $LOG
		exit 1
fi

echo -e "Verificando conexão com internet..."
if [ "$(nc -zw1 google.com 443 &> /dev/null ; echo $?)" == "0" ]
	then
		echo -e "Você tem acesso a Internet, continuando com o script..." >> $LOG
		sleep 5
	else
		echo -e "Você NÃO tem acesso a Internet, verifique suas configurações de rede IPV4 e conexões e execute novamente este script." >> $LOG
		sleep 5
		exit 1
fi

echo "Passou pelas verificações iniciais!" &>> $LOGD
sleep 5

#============================================================================================
#* ATUALIZAÇÃO DO SISTEMA
HORA_UP=$(date +%T)
echo -e "Atualizando repositórios..."
echo "Atualizando repositórios..." >> $LOG
echo "Atualizando repositórios..." >> $LOGD
if ! sudo apt-get update &>> $LOGD
then
    echo -e "Não foi possível atualizar os repositórios. Verifique seu arquivo /etc/apt/sources.list"
    echo "Não foi possível atualizar os repositórios. Verifique seu arquivo /etc/apt/sources.list" >> $LOG
    exit 1
fi
echo -e "Atualização do repositório feita com sucesso!!"

echo -e "Atualizando pacotes e sistema..."
echo "Atualizando pacotes e sistema..." >> $LOG
echo "Atualizando pacotes e sistema..." >> $LOGD
if ! sudo apt-get upgrade --autoremove -y &>> $LOGD
then
    echo -e "Não foi possível atualizar pacotes."
    echo "Não foi possível atualizar pacotes." >> $LOG
    exit 1
fi
echo -e "Atualização de pacotes feita com sucesso!!"
HORA_UPFIM=$(date +%T)

#============================================================================================
#* INSTALAÇÂO DE SOFTWARES .DEB ADMINISTRATIVOS
HORA_DEB=$(date +%T)
if ! sudo apt-get install $WSL2DEB -y &>> $LOGD
then
	echo -e "Não foi possivel instalar todos os pacotes DEB." >> $LOG
	echo -e "Verifique o LOG em $LOGD para maiores detalhes.."
	echo "Não foi possivel instalar todos os pacotes DEB." >> $LOGD
	exit 1
fi
echo -e "Instalação dos pacotes DEB finalizados com sucesso!!"
# Instalação do acme para gerar crertificados
echo -e "Download e instalação do ACME..."
sudo wget -O - https://get.acme.sh | sh &>> $LOGD
HORA_DEBFIM=$(date +%T)

HORAFINAL=$(date +%T)

#============================================================================================
#* CALCULOS DOS TEMPO DE EXECUÇÃO
HORAUP01=$(date -u -d "$HORA_UP" +"%s")
HORAUP02=$(date -u -d "$HORA_UPFIM" +"%s")
TEMPO_UP=$(date -u -d "0 $HORAUP01 sec - $HORAUP02 sec" +"%H:%M:%S")
echo -e "Tempo gasto para atualização do sistema: $TEMPO_UP" >> $LOG
echo "Tempo gasto para atualização do sistema: $TEMPO_UP" &>> $LOGD

HORAINICIAL01=$(date -u -d "$HORAINICIAL" +"%s")
HORAFINAL01=$(date -u -d "$HORAFINAL" +"%s")
TEMPO=$(date -u -d "0 $HORAFINAL01 sec - $HORAINICIAL01 sec" +"%H:%M:%S")
echo -e "Tempo gasto para execução do script $0: $TEMPO" >> $LOG
echo "Tempo gasto para execução do script $0: $TEMPO" &>> $LOGD